### PR TITLE
Add interrupt generating command

### PR DESCRIPTION
**Pull request by adamgreig**
This moves `interrupts.py` from stm32-rs to svdtools, albeit modified to print to stdout and only process one svd file at a time, which seems more easily batched/redirected/etc.

Printing interrupt tables is useful to check that all interrupts you expect are present and in the correct peripheral, and to highlight gaps that might indicate some interrupts are missing from the SVD.

### CODE REVIEW

File: svdtools/interrupts.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+def parse_device(svd_file):
+    interrupts = {}
+    tree = ET.parse(svd_file)
+    dname = tree.find("name").text
+    for ptag in tree.iter("peripheral"):
+        pname = ptag.find("name").text
+        for itag in ptag.iter("interrupt"):
+            name = itag.find("name").text
+            value = itag.find("value").text
+            desc = itag.find("description").text.replace("\n", " ")
```
Comment (nickray) [inline svdtools/interrupts.py]: Again for LPC55, these descriptions seem to be missing.

Proposed patch:
```
            maybe_desc = itag.find("description")    
            desc = maybe_desc.text.replace("\n", " ") if maybe_desc is not None else ""  
```

Comment (adamgreig) [reply svdtools/interrupts.py]: Thanks, done!

**Comment by adamgreig** [comment]
bors retry

**Comment by nickray** [comment]
Was this my mistake or a bors issue again? I see you added a commit after the "Merge conflict" and am not sure I follow.

**Comment by adamgreig** [comment]
Neither - my two PRs were both based on the same master branch, but both modify e.g. `__init__.py` to change what's imported, so git can't automatically merge them. Once bors merged the first PR, it couldn't merge the second PR into the new master, so gives the merge conflict message. I manually merged master into the PR, and then bors is able to cleanly merge the PR into master.

Not much to be done about two PRs both changing the same file in different ways - I could have done one PR based on top of the other but then it's much harder to see what's unique to each PR. I could have rebased this one after the first one merged, but the rebase was a bit annoying since it conflicts at each commit, so a merge seemed easier.

**Comment by nickray** [comment]
Got it. I think the merge status of this updated after the first merge, so I should have waited for the first to complete. Got it now! Nice thing bors caught it though!

**Pull request merged**